### PR TITLE
Adding NeRF-OSR Dataset

### PR DESCRIPTION
**Pull request by JADGardner**
Adding a dataparser and download script for the [NeRF-OSR](https://4dqv.mpi-inf.mpg.de/NeRF-OSR/) dataset. The images in the dataset are of varying resolutions, so the VariableResDataManagerConfig is currently required.

image

### CODE REVIEW

File: nerfstudio/data/dataparsers/nerfosr_dataparser.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    ----------
+    directory : str
+        The directory to search for files.
+    exts : list of str
+        A list of file extensions to search for. Each file extension should be in the form '*.ext'.
+
+    Returns
+    -------
+    list of str
+        A list of file paths for all the files that were found. The list is sorted alphabetically.
```
Comment (tancik) [inline nerfstudio/data/dataparsers/nerfosr_dataparser.py:54]: Should be in the form to match the rest of the codebase,
```
"""Find all files in a directory that have a certain file extension.

Args:
    directory : The directory to search for files.
    exts :  A list of file extensions to search for. Each file extension should be in the form '*.ext'.

Returns:
    A list of file paths for all the files that were found. The list is sorted alphabetically.
 ```

File: nerfstudio/data/dataparsers/nerfosr_dataparser.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        files_grabbed = []
+        for ext in exts:
+            files_grabbed.extend(glob.glob(os.path.join(directory, ext)))
+        if len(files_grabbed) > 0:
+            files_grabbed = sorted(files_grabbed)
+        return files_grabbed
+    return []
+
+
+def parse_txt(filename):
```
Comment (tancik) [inline nerfstudio/data/dataparsers/nerfosr_dataparser.py:67]: `parse_txt` -> `_parse_osm_txt`

**Review by tancik — COMMENTED** [review]
Looks like a cool dataset, thanks for adding!

File: nerfstudio/data/dataparsers/nerfosr_dataparser.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    DataParser,
+    DataParserConfig,
+    DataparserOutputs,
+)
+from nerfstudio.data.scene_box import SceneBox
+
+CONSOLE = Console(width=120)
+
+
+def find_files(directory, exts):
```
[the remark this answers is not in the document]
Comment (JADGardner) [reply nerfstudio/data/dataparsers/nerfosr_dataparser.py]: Done.

**Review by tancik — APPROVED** [review]
LGTM

**Pull request merged**